assets/js/hooks/useMonitorInternetConnection.js

```diff
@@ -53,15 +54,9 @@ export function useMonitorInternetConnection() {
 		}
 
 		try {
-			const connectionCheckResponse = await apiFetch( {
-				path: '/google-site-kit/v1/',
-			} );
-
-			// We are only interested if the request was successful, to
-			// confirm online status.
-			const canReachConnectionCheck = !! connectionCheckResponse;
+			await fetch( new URL( '/google-site-kit/v1/', rootURL ) );
```

Comment:
After some more thought here, I think this is not quite the right solution (criticizing my own IB here! 😄).

The fact that we're reaching for some internal api-fetch data here is a smell that we should be using that instead. The reason I suggested using `fetch` instead was to avoid the additional cases where `api-fetch` throws, where `fetch` only throws when the request fails. Essentially we should be able to do the same using `api-fetch` with an additional check in the throw block as `api-fetch` returns a specific error in this case. See https://github.com/WordPress/gutenberg/blob/d9b726b8451746703cc1b9680487e3726ab4a03f/packages/api-fetch/src/index.js#L130-L131

In summary, we can keep most of what we had before, but adjusted to use an approach closer to the suggested one here in letting the try/catch determine the result. The difference in using `api-fetch` is that we need to set the online state conditionally depending on the error code if it throws. I.e. if it throws and the error code is `fetch_error`, only then should we set it to be offline. Any other error would mean the request happened, which is all we are concerned about here, and can be ignored.